src/shared/tools.h

```diff
@@ -140,6 +141,16 @@ static inline int bitscan(uint mask)
 #define PRINTFARGS(fmt, args)
 #endif
 
+template<class T>
+const bool inrange(std::vector<T> vec, size_t i) {
+    return i < vec.size();
```

Comment:
Do these functions really add any useful functionality? I don't think so, honestly.

Reply:
It helps replacing the existing custom vector with std::vector though